src/gan/gen_pyt/datasets/english/dataset.py

```diff
@@ -0,0 +1,247 @@
+# coding=utf-8
+
+import re
+import numpy as np
+import pickle
+from sklearn.model_selection import train_test_split
+
+from multivac.src.gan.gen_pyt.asdl.hypothesis import *
+from multivac.src.gan.gen_pyt.asdl.lang.eng.eng_asdl_helper \
+    import english_ast_to_asdl_ast, asdl_ast_to_english_ast
+from multivac.src.gan.gen_pyt.asdl.lang.eng.eng_transition_system \
+    import EnglishTransitionSystem
+
+from multivac.src.gan.gen_pyt.components.action_info \
+    import ActionInfo, get_action_infos
+from multivac.src.gan.gen_pyt.components.dataset import Example
+from multivac.src.gan.gen_pyt.components.vocab import Vocab, VocabEntry
+
+from multivac.src.gan.gen_pyt.utils.io_utils \
+    import serialize_to_file, deserialize_from_file
+
+from multivac.src.rdf_graph.rdf_parse import tokenize_text
+
+class English(object):
+    @staticmethod
+    def canonicalize_example(query, text, parser):
+        '''
+        If we want to do any cleaning on the text, here's where to do it.
+        '''
+        try:
+            query = stanford_parse(parser, query)
+        except:
+            print('Could not parse query: {}'.format(query))
+            return None, None, None
+
+        try:
+            parse_tree = english_ast_to_asdl_ast(query.parse_string)
+        except: 
+            print("Could not interpret query parse: {}".format(query.parse_string))
+            return None, None, None
+
+        query_tokens = [x.text for x in query.tokens]
+        parse_str = query.parse_string
+
+        return query_tokens, text, parse_str, parse_tree
+
+    @staticmethod
+    def check_parse(query):
+        parseable = True
+
+        if any([x.pos_.upper().startswith('VB') for x in query.tokens]):
+            pass
+        else:
+            parseable = False
+
+        return parseable
+
+    @staticmethod
+    def clean_queries(queries, verbose=False):
+        clean = list()
+
+        if verbose:
+            print(("Performing basic clean on {} queries.".format(len(queries))))
+
+        for query in queries:
+            # strip whitespace, and quotes
+            # Remove any sentence fragments preceding question
+            # Remove non-alphabetic characters at the start of the string
+            query = query.strip()
+            query = re.sub(r"“|”", "\"", query)
+            query = re.sub(r"‘|’", "\'", query)
+            query = re.sub(r"`", "\'", query)
+            query = query.strip("\"")
+            query = query.strip("\'")
+            query = query[query.index(re.split(r"\"", query)[-1]):]
+            query = query[query.index(re.split(r"NumericCitation", query, re.IGNORECASE)[-1]):]
+            query = query[query.index(re.split(r"[\.\!\?]\s+", query)[-1]):]
```

Comment:
Some of these could benefit from a comment above talking about what it's removing. figuring it out from the pattern is a little tough at a quick glance.